lua/neogit/buffers/commit_select_view/init.lua

```diff
@@ -98,9 +96,14 @@ function M:open(action)
         end
       end,
     },
-    after = function()
-      if commit_at_cursor then
-        vim.fn.search(commit_at_cursor.oid)
+    after = function(buffer, win)
+      if item and item.commit then
```

Comment:
Not opposed to this change, I'd just recommend setting the number of commits to something fairly high (couple thousand) and see if/how it impacts the buffer load time. Maybe its negligible, just wanna check.

The find_component function doesn't have any kind of early return once its found something iirc, so it's going to loop over every single component even if it finds it on the first go.

Same goes for the log buffer.

Reply:
Perhaps we should just short circuit in the find_component then, and add an option for max_count in it